src/lib/ptree/json.lua

```diff
@@ -20,13 +35,20 @@ local function skip_whitespace(input)
    drop_while(input, whitespace_pat)
 end
 
-local function check(input, ch)
+local function peek(input, ch)
```

Comment:
I think `peek` should only peek a char and return true or false depending whether there was a match. 

```lua
local function peek(input, ch)
   return input:peek_char() == ch
end 
```

Then if there was a match you will run `consume` or `check`. This is in line with what other parsers do in the code base (for instance, `lib/yang/parser.lua`).